dimod/bqm/utils.pyx

```diff
@@ -40,6 +42,53 @@ cdef object as_numpy_scalar(double a, np.dtype dtype):
     """Note that the memory is interpreted to match dtype, not a cast"""
     return PyArray_Scalar(&a, dtype, None)
 
+def cylmin(cyBQM bqm):
+    cdef double min_ = sys.maxsize
```

Comment:
Same for `cyqmin()`.

Reply:
This will currently return `sys.maxsize` for an empty BQM, which I think is very confusing behavior. I would rather follow python's behavior and fail when `bqm.num_interactions == 0`, but allow the user to provide a default.

NumPy has a [similar notion](https://numpy.org/doc/stable/reference/generated/numpy.amin.html#numpy.amin).

Reply:
I like NumPy's `initial`.